### PR TITLE
fix/PARA-21338/hoop-redis-auth

### PR DESCRIPTION
**Pull request by franklingarcia-paragon**
### Issues Closed

- [PARA-21338](https://useparagon.atlassian.net/browse/PARA-21338)

### Brief Summary

hoop redis auth from infra-output

### Detailed Summary

Hoop Redis connections were provisioned with host, port, TLS, and CA only—no credentials—even when Azure/GCP infra outputs include a Redis password (required for on Azure Cache for Redis). This change extends the Hoop module's `infra_vars.redis` schema and wires optional `password` and `user` into connection secrets as `envvar:PASS` and `envvar:USER`, matching the existing Postgres pattern. When those fields are absent (typical AWS ElastiCache), behavior is unchanged.

### Changes

- Added optional `password` and `user` to `infra_vars.redis` in `hoop/variables.tf` (AWS, Azure, GCP)
- Conditionally set `envvar:PASS` and `envvar:USER` on Hoop Redis connections in `hoop/locals.tf` when values are present in `infra-output.json` (AWS, Azure, GCP)

### Unrelated Changes

- None

### Future Work

- Align `locals.tf` null checks with the `ca_certificate` pattern (`!= null && != ""`) to avoid edge cases when `password` is explicitly null
- Smoke-test on  after apply; if Hoop/redis-cli does not pick up `PASS`/`USER`, add `envvar:REDISCLI_AUTH` as a fallback
- Add Azure `username` to infra outputs only if a customer requires Redis ACL with a non-default user

### Steps to Test

1. `cd azure/workspaces/paragon && terraform init -backend=false && terraform validate` (repeat for `aws/` and `gcp/` if desired)
2. On an Azure deployment with Redis auth, ensure `infra-output.json` includes `redis.*.password` from infra apply
3. Run `terraform plan` / apply on the `paragon` workspace with `hoop_enabled = true`
4. In Hoop, open a Redis connection (e.g. `{prefix}-redis-cache`) and run `PING` or a read-only command—should succeed without `NOAUTH`
5. On AWS (no Redis password in infra output), confirm Hoop Redis connections are unchanged (no `envvar:PASS` in plan)

### QA Notes

- Credentials come from existing sensitive infra outputs; they should not appear in non-sensitive Terraform logs
- Functional validation requires Hoop access to the target environment after apply—local `terraform validate` alone is insufficient

### Deployment Notes

- No new tfvars or Helm changes; requires `paragon` workspace apply so Hoop connections are updated
- Azure: confirm `infra-output.json` already includes `password` under `redis` (from `azure/workspaces/infra/redis/outputs.tf`)

### Screenshots

- N/A (Terraform / Hoop connection config only)

[PARA-21338]: https://useparagon.atlassian.net/browse/PARA-21338?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: azure/workspaces/paragon/hoop/variables.tf [hunk 1]
```diff
@@ -127,6 +127,8 @@ variable "infra_vars" {
         db_number      = optional(number, 0)
```
Comment (tedparagon) [inline azure/workspaces/paragon/hoop/variables.tf:128]: This variable should be flagged as sensitive if it is going to hold a password.

File: aws/workspaces/paragon/hoop/variables.tf [hunk 2]
```diff
@@ -127,6 +127,8 @@ variable "infra_vars" {
         db_number      = optional(number, 0)
```
Comment (tedparagon) [inline aws/workspaces/paragon/hoop/variables.tf:128]: This variable should be flagged as sensitive if it is going to hold a password.

File: gcp/workspaces/paragon/hoop/variables.tf [hunk 3]
```diff
@@ -127,6 +127,8 @@ variable "infra_vars" {
         db_number      = optional(number, 0)
```
Comment (tedparagon) [inline gcp/workspaces/paragon/hoop/variables.tf:128]: This variable should be flagged as sensitive if it is going to hold a password.

**Review by richard-paragon — APPROVED** [review]

**Review by tedparagon — APPROVED** [review]
LGTM
